pkg/v18/resource/tiller/resource.go

```diff
@@ -125,26 +125,26 @@ func (r *Resource) ensureTillerInstalled(ctx context.Context, obj interface{}) e
 		r.logger.LogCtx(ctx, "level", "debug", "message", "timeout fetching certificates")
 
 		// A timeout error here means that the cluster-operator certificate
-		// for the current guest cluster was not found. We can't continue
+		// for the current tenant cluster was not found. We can't continue
 		// without a Helm client. We will retry during the next execution, when
 		// the certificate might be available.
 		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
 		reconciliationcanceledcontext.SetCanceled(ctx)
 
 		return nil
-	} else if helmclient.IsTillerInstallationFailed(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "Tiller installation failed")
+	} else if helmclient.IsTillerNotFound(err) {
```

Comment:
The error we matched on before wasn't even used. 🤦‍♂

Reply:
How did you find that? can we have a unittest for this?

Reply:
I found it during manual testing. I'm not sure how we can catch that in a unit test.